fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -560,7 +563,7 @@ def local_state_dict(self, *args, **kwargs):  # type: ignore
             for module in self.modules():  # includes self
                 if isinstance(module, FullyShardedDataParallel):
                     stack.enter_context(module._no_return_full_state_dict())
-            return self.state_dict(*args, **kwargs)
+            return self._state_dict(*args, **kwargs)
```

Comment:
I think you don't need to introduce the `_state_dict` function if you do the following:

`FSDP.state_dict(self, *args, *kwargs)`

This ensures that you will call the right version state_dict from this function. It should be the same as the changes you have done here and should be simpler.

I don't know if this is consider pythonic though. Seems a like simpler and clearer method to me.

Reply:
Since `state_dict()` is an object level method and not a class method, `FSDP.state_dict()` won't work in this case.

I had hoped `super().state_dict()` calls the fairscale version. But in practice, it doesn't call the fairscale version as explained in the PR description.
I will try to dig into why that might be the case.

Reply:
Have you tried `FSDP.state_dict()`? It should work since you give the `self` argument explicitly. I think doing that would explicitly call the right version of it.